server/app/main.py

```diff
@@ -1,76 +1,73 @@
 from fastapi import FastAPI, HTTPException, Depends
+
 from sqlalchemy.orm import Session
-from app import utils, db
 from pydantic import BaseModel
+from typing import Optional
+
+from app import db
+
+from services.client_factory import ClientFactory
+
+from utilities.utility_functions import * 
+from utilities.constants.LLM_enums import LLMType, ModelType
+from utilities.constants.prompts_enums import PromptType
+from utilities.constants.response_messages import ERROR_QUESTION_REQUIRED, ERROR_SHOTS_REQUIRED
+from utilities.prompts.prompt_factory import PromptFactory
 
 app = FastAPI()
 
 # Pydantic models for request body validation
 class QueryRequest(BaseModel):
     query: str
 
-class QuestionRequest(BaseModel):
+class QueryGenerationRequest(BaseModel):
     question: str
+    prompt_type: PromptType = PromptType.SQL_ONLY
+    shots: Optional[int] = None
+    llm_type: Optional[LLMType] = LLMType.OPENAI
+    model: Optional[ModelType] = ModelType.GPT4_O
```

Comment:
We can rename the model names as "OPENAI_GPT4_O", "OPENAI_GPT4", "OPENAI_GPT4_O_MINI",